### ISSUE TITLE
Crash report 8.1: NPE in PostStore.updatePost()

### ISSUE THREAD

**Issue by aforcier** [wordpress-mobile/WordPress-Android#6530]
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void org.wordpress.android.fluxc.model.PostModel.setDateLocallyChanged(java.lang.String)' on a null object reference
       at org.wordpress.android.fluxc.store.PostStore.updatePost(PostStore.java:548)
       at org.wordpress.android.fluxc.store.PostStore.onAction(PostStore.java:365)
       at java.lang.reflect.Method.invoke(Method.java)
       at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
```

The crash is in FluxC, but we should understand why we're ending up with a `null` `PostModel` when sending an `UPDATE_POST` action from WPAndroid.

**Comment by mzorz** [comment]
Managed to reproduce this fairly consistently when adding lots of images to a post and then immediately exiting the editor then tapping on `DELETE/TRASH`. When the post is deleted, it is deleted _ before_  the signal to cancel uploads properly reaches the `UploadService`, and thus when it does, the service tries to update the Post with the "failed"  media uploads first, but the Post doesn't exist anymore (and we're not checking so), thus the crash happens.

Will write a PR to make `updatePostModelWithCompletedAndFailedUploads` check for null before trying to update the Post.

**Comment by aforcier** [comment]
Fixed in #6534.
